sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -626,6 +618,32 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
     }
   }
 
+  private def grow(neededSize: Int): Unit = {
+    // There is 8 bytes for the pointer to next value
+    val totalNeededSize = cursor + 8 + neededSize
```

Comment:
nit:
```
private def grow(inputRowSize: Int): Unit = {
  val neededSize = cursor + 8 + inputRowSize
  val currentSize = page.length * 8L + Platform.LONG_ARRAY_OFFSET
  ...
}
```

Reply:
@cloud-fan Thank you for your suggestion and code.